test/performance/odp_timer_accuracy.c

```diff
@@ -1449,6 +1642,41 @@ int main(int argc, char *argv[])
 			odp_time_wait_ns(10 * ODP_TIME_MSEC_IN_NS);
 
 		cancel_periodic_timers(test_global);
+	} else if (test_global->opt.mode == MODE_CONCURRENCY) {
+		uint64_t start_ns = odp_time_global_strict_ns();
+		uint64_t prev_ns = start_ns;
+		uint64_t ep[ODP_THREAD_COUNT_MAX] = { 0 };
+		uint64_t events_prev = 0;
+		uint64_t events = 0;
+
+		while ((events = odp_atomic_load_u64(&test_global->events)) <
+		       test_global->opt.tot_timers) {
+			odp_time_wait_ns(test_global->opt.interval * ODP_TIME_SEC_IN_NS);
+
+			uint64_t ns = odp_time_global_strict_ns();
+			uint64_t msec = (ns - prev_ns) / ODP_TIME_MSEC_IN_NS;
+			uint64_t msec_total = (ns - start_ns) / ODP_TIME_MSEC_IN_NS;
+			uint64_t e = events - events_prev;
+			int stopped = 0;
+
+			for (int i = 0; i < (int)test_global->opt.burst; i++) {
+				/*
+				 * Count how many timers have not generated any events since last
+				 * interval.
+				 */
+				timer_ctx_t *ctx = &test_global->timer_ctx[i];
+				uint64_t ctx_events = ctx->events;
```

Comment:
Seems like a possible data race. Is this information that necessary to print anyway?

Reply:
It's definitely a data race. It's avoiding synchronization in order to have minimal effect on the test itself. The prints have been very useful in recent testing, especially when some timers stop generating events, some periodic check like this is needed to see that something stopped from happening.

On the other hand perhaps it's better to add prints ad hoc, when needed to investigate a specific problem. So this could just be removed.

Reply:
Looks like `timer_ctx_t.events` could be operated with relaxed atomic, so the overhead should be relatively small. However, either solution is fine for me.

Reply:
Relaxed atomic would probably be fine, but nevertheless I decided to remove the "stopped" counter, so ctx->events remains non-atomic.